react-ui/src/pages/analyzeVotes/PearsonsCorrelation.jsx

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable max-len */
+/* eslint-disable react/forbid-prop-types */
+import PropTypes from 'prop-types';
+import { useCallback, useEffect, useMemo } from 'react';
+import Plot from 'react-plotly.js';
+
+import {
+  roundTwoDecimals, createTraces, trimUsername,
+} from './functions';
+import MARKERS from './markers';
+
+const GROUP = { selected: 0, none: 1, few: 2 };
+
+/**
+ * Compare user similarity
+ */
+function PearsonsCorrelation({
+  username, username2, votes, entryPositionLookup, setUsername2, voteMinimum,
+}) {
+  /**
+   * CALCULATE PEARSONS
+   */
+  const pearsons = useMemo(() => {
+    // USER DOES NOT EXISTS
+    if (!username) { return []; }
+
+    const numberOfEntries = Object.keys(entryPositionLookup).length;
+    const allVotesByUser = votes.reduce((acc, vote) => {
+      if (!acc[vote.username]) {
+        acc[vote.username] = new Array(numberOfEntries);
+      }
+      acc[vote.username][entryPositionLookup[vote.entryId]] = vote.rating;
+      return acc;
+    }, {});
+
+    const userVotes = allVotesByUser[username];
+    // USER DID NOT VOTE
+    if (!userVotes) { return []; }
+
+    return Object.keys(allVotesByUser).filter((u) => u !== username).map((u) => {
+      const otherUser = allVotesByUser[u];
+      let sum1 = 0; let sum2 = 0; let sum1Sq = 0; let sum2Sq = 0; let sumProduct = 0;
```

Comment:
_Optional_: You could do `let sum1 = 0, sum2 = 0, ...` instead, if you want to put them all on one line.

Reply:
That was my original plan, but my eslint gets mad at me when I do that.